include/openthread/dnssd_server.h

```diff
@@ -0,0 +1,180 @@
+ * active subscriptions and stop notifying when there is no active subscription for @p aFullName.
+ *
+ * @param[in] aContext      A pointer to the application-specific context.
+ * @param[in] aFullName     The null-terminated full service name (e.g. "_ipps._tcp.default.service.arpa."),
+ *                          or full service instance name (e.g. "OpenThread._ipps._tcp.default.service.arpa."),
+ *                          or full host name (e.g. "ot-host.default.service.arpa.").
+ *
+ * @sa otDnssdQueryHandleDiscoveredServiceInstance
+ * @sa otDnssdQueryHandleDiscoveredHost
+ *
+ */
+typedef void (*otDnssdQuerySubscribeCallback)(void *aContext, const char *aFullName);
+
+/**
+ * This function is called when a DNS-SD query unsubscribes either of:
+ *      1. a service name.
+ *      2. a service instance name.
+ *      3. a host name.
+ *
+ * The DNS-SD query implementation is responsible for identifying what @p aFullName is.
+ *
+ * @note There can be multiple subscription to the same name. DNS-SD query implementation should  record the number of
+ * active subscriptions and stop notifying when there is no active subscription for @p aFullName.
+ *
+ * @param[in] aContext      A pointer to the application-specific context.
+ * @param[in] aFullName     The null-terminated full service name (e.g. "_ipps._tcp.default.service.arpa."), or
+ *                          full service instance name (e.g. "OpenThread._ipps._tcp.default.service.arpa.").
+ *
+ */
+typedef void (*otDnssdQueryUnsubscribeCallback)(void *aContext, const char *aFullName);
+
+/**
+ * This structure represents information of a discovered service instance for a DNS-SD query.
+ *
+ */
+typedef struct otDnssdServiceInstanceInfo
+{
+    const char *        mFullName;   ///< Full instance name (e.g. "OpenThread._ipps._tcp.default.service.arpa.").
+    const char *        mHostName;   ///< Host name (e.g. "ot-host.default.service.arpa.").
+    uint8_t             mAddressNum; ///< Number of host IPv6 addresses.
+    const otIp6Address *mAddresses;  ///< Host IPv6 addresses.
+    uint16_t            mPort;       ///< Service port.
+    uint16_t            mPriority;   ///< Service priority.
+    uint16_t            mWeight;     ///< Service weight.
+    uint16_t            mTxtLength;  ///< Service TXT RDATA length.
+    const uint8_t *     mTxtData;    ///< Service TXT RDATA.
+    uint32_t            mTtl;        ///< Service TTL (in seconds).
+} otDnssdServiceInstanceInfo;
+
+/**
+ * This structure represents information of a discovered host for a DNS-SD query.
+ *
+ */
+typedef struct otDnssdHostInfo
+{
+    uint8_t             mAddressNum; ///< Number of host IPv6 addresses.
+    const otIp6Address *mAddresses;  ///< Host IPv6 address.
+    uint32_t            mTtl;        ///< Service TTL (in seconds).
+} otDnssdHostInfo;
+
+/**
+ * This function sets DNS-SD server query callbacks. The DNS-SD server calls @p aSubscribe to subscribe to a service or
```

Comment:
nit: first sentence is the brief - keep in its own block. Add an empty line after the first sentence. Other comment blocks should be updated as well.